src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -105,6 +180,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     return recipesMatchingList;
   }
 
+  /**
+   * Returns a list of matched ingredients snippet by the search query.
+   */
+  private ArrayList<String> returnMatchedIngredients(ScoredDocument document) {
+    ArrayList<String> ingredientsMatched = new ArrayList<>();
+
+    List<Field> listFields = document.getExpressions();
```

Comment:
fields

don't repeat the type name in the variable name, it's clear from the definition.